hatch.toml

```diff
@@ -6,7 +6,7 @@ pre-install-commands = [
 [envs.default.scripts]
 sync = "pip install -r requirements-testing.txt"
 test = "pytest --cov-config pyproject.toml {args:test}"
-typing = "mypy {args:src test}"
+typing = "mypy --implicit-optional {args:src test}"
```

Comment:
Yeah, we have a lot of Optional[str] parameters in OpenJob entitiy constructors (`__init__`) and linter raised errors on `@property` methods and in all usages even if we add the check for `None`. It would be great if you could recommend a best practice for this that you use to ensure that your code style is followed.

Reply:
If you're declaring everything as (for example) Optional[str] = None rather than just str = None you're still running into issues?  Can you point me towards an example spot?  I just tried updating the last error I saw in hatch run typing which was in adaptor.py with exit_code in _record_error_and_raise and it seemed to fix that one at least.

I haven't tried it personally but mypy seems to suggest https://github.com/hauntsaninja/no_implicit_optional as a way to bulk update the code rather than have to go through everything individually